test/test_tokenizer.rb

```diff
@@ -170,9 +170,10 @@ def test_operators
     assert_equal %w(~ a), tokenize("~a")
 
     # Edge cases
-    assert_equal %w(-!#$%&*+,.:;<=>?@\\^_`|~), tokenize("-!#$%&*+,.:;<=>?@\\^_`|~")
-    assert_equal %w(-!#$%&*+,.:;<=>?@\\^_`|~*/), tokenize("-!#$%&*+,.:;<=>?@\\^_`|~*/")
-    assert_equal %w(/-!#$%&*+,.:;<=>?@\\^_`|~), tokenize("/-!#$%&*+,.:;<=>?@\\^_`|~")
+    assert_equal %w(-!#$%&*+,.:;<=>), tokenize("-!#$%&*+,.:;<=>")
+    assert_equal %w(-!#$%&?@\\^_`|~), tokenize("-!#$%&?@\\^_`|~")
+    assert_equal %w(-!#$%&*+,.:;<=>), tokenize("-!#$%&*+,.:;<=>")
+    assert_equal %w(/-!#$%&*+,.:;<>), tokenize("/-!#$%&*+,.:;<>")
```

Comment:
🤔 shouldn't these tests be confirming the truncation behaviour too ie keep the original argument to `tokenize()` but update the assertion?

Reply:
There's a new test below testing for truncation. This one is meant to test that these characters do not produce undesired token splits.

Reply:
Yeah, I saw that, but I was wondering if we wanted to check the truncation here too. We'll leave it as-is to keep the test purpose separate.